File: Python/Iniciante/1040.py
n1, n2, n3, n4 = input().split()
n1 = float(n1)
n2 = float(n2)
n3 = float(n3)
n4 = float(n4)

p1 = 2
p2 = 3
p3 = 4
p4 = 1

media = (n1*p1 + n2*p2 + n3*p3 + n4*p4) / (p1+p2+p3+p4)

if media >= 7.0: 
  print(f"Media: {media:.1f}")
  print("Aluno aprovado.")
  exame = False
  
elif media < 5.0: 
  print(f"Media: {media:.1f}")
  print("Aluno reprovado.")
  exame = False
  
elif 5.0 <= media <= 6.9: 
  print(f"Media: {media:.1f}")
  print("Aluno em exame.")
  exame = True

if exame:
  n5 = float(input())
  print(f"Nota do exame: {n5:.1f}")

  media = (media + n5) / 2

  if media >= 5.0: 
    print("Aluno aprovado.")
  else: 
    print("Aluno reprovado.")
  
  print(f"Media final: {media:.1f}")

File: Java/Iniciante/1040.java
import java.io.IOException;
import java.util.Scanner;

public class Main {

  public static void main(String[] args) throws IOException {
    Scanner input = new Scanner(System.in);

    float n1 = input.nextFloat();
    float n2 = input.nextFloat();
    float n3 = input.nextFloat();
    float n4 = input.nextFloat();

    float media = ((n1 * 2) + (n2 * 3) + (n3 * 4) + n4) / 10;

    System.out.printf("Media: %.1f\n", media);

    if(media >= 7){
      System.out.println("Aluno aprovado.");
    }
    else if(media >= 5 && media <= 6.9){
      System.out.println("Aluno em exame.");
      float n5 = input.nextFloat();
        
      System.out.printf("Nota do exame: %.1f\n", n5);

      media = (media + n5) / 2;

      if(media >= 5){
        System.out.println("Aluno aprovado.");
        System.out.printf("Media final: %.1f\n", media);
      }
      else{
        System.out.println("Aluno reprovado.");
      }
      
    }
    else{
      System.out.println("Aluno reprovado.");
    }
  }
}
